plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -222,9 +223,9 @@ type CekEvaluationException uni fun = CekEvaluationExceptionCarrying (Term Name
 type CekM cost uni fun s = CekCarryingM cost (Term Name uni fun ()) uni fun s
 
 -- | A budgeting mode to execute an evaluator in.
-data ExBudgetMode cost uni fun = ExBudgetMode
-    { _exBudgetModeSpender :: CekBudgetSpender cost uni fun  -- ^ A spending function.
-    , _exBudgetModeInitSt  :: cost                           -- ^ An initial state.
+data ExBudgetMode cost uni fun s = ExBudgetMode
+    { _exBudgetModeSpender  :: CekBudgetSpender cost uni fun s  -- ^ A spending function.
+    , _exBudgetModeGetFinal :: ST s cost                        -- ^ For accessing the final state.
```

Comment:
This actually lets you get the state at any time, right? It's just a way to get the state (which we happen to only use at the end).